### PR TITLE
Add rust-client to Nix flake and NixOS module

### PR DESCRIPTION
**Pull request by MarcoPolo**
A continuation of https://github.com/dennis-tra/punchr/pull/26 to add the rust-client to the NixOS module. Also lets Nix users run the rust client with the one-liner:

```
API_KEY="..." nix run github:marcopolo/punchr/marco/add-rust-client-to-nix#rust-client
# Or when this is merged: 
# API_KEY="..." nix run github:dennis-tra/punchr#rust-client
```

Adds a CI job to make sure the code builds.

### CODE REVIEW

**Comment by MarcoPolo** [comment]
Rust builds are currently broken though, so not much use right now :/

File: .gitignore [hunk 1]
```diff
@@ -196,10 +196,6 @@ Temporary Items
 debug/
 target/
 
-# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
-# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
-Cargo.lock
-
```
Comment (elenaf9) [inline .gitignore:202]: Not against tracking the Cargo.lock, but still wondering: do you have a specific reason for adding it?

**Review by elenaf9 — APPROVED** [review]
Can't really comment on the nix module but I trust Marco knows what he is doing :smile:.

I plan on working on #37 towards the end of this week, sorry for the delay. Help is welcomed in case anybody wants to work on this before I get to it.

File: .gitignore (hunk 1 above)
Comment (MarcoPolo) [reply .gitignore:202]: Yes. Without the Cargo.lock file this build isn't reproducible. Nix is all about reproducible builds. Cargo.lock should generally be included for applications and not for libraries.

From the link referenced here:
> If you’re building a non-end product, such as a rust library that other rust [packages](https://doc.rust-lang.org/cargo/appendix/glossary.html#package) will depend on, put Cargo.lock in your .gitignore. If you’re building an end product, which are executable like command-line tool or an application, or a system library with crate-type of staticlib or cdylib, check Cargo.lock into git.

we're building an application here.

Comment (mxinden) [reply .gitignore:202]: > Cargo.lock should generally be included for applications and not for libraries.

:+1:

**Comment by MarcoPolo** [comment]
Got the rust-client building thanks to a quick pairing session with @mxinden, I just had to update the rustc version I was using :)

**Pull request merged**